src/lib/operators/abstract_aggregate_operator.hpp

```diff
@@ -0,0 +1,119 @@
+#pragma once
+
+#include "expression/aggregate_expression.hpp"
+#include "operators/abstract_operator.hpp"
+#include "operators/abstract_read_only_operator.hpp"
+#include "type_comparison.hpp"
+#include "types.hpp"
+
+namespace opossum {
+
+/**
+ * Aggregates are defined by the column (ColumnID for Operators, LQPColumnReference in LQP) they operate on and the aggregate
+ * function they use. COUNT() is the exception that doesn't use a column, which is why column is optional
+ * Optionally, an alias can be specified to use as the output name.
+ */
+struct AggregateColumnDefinition final {
+  AggregateColumnDefinition(const std::optional<ColumnID>& column, const AggregateFunction function)
+      : column(column), function(function) {}
+
+  std::optional<ColumnID> column;
+  AggregateFunction function;
+};
+
+/*
+The AggregateFunctionBuilder is used to create the lambda function that will be used by
+the AggregateVisitor. It is a separate class because methods cannot be partially specialized.
+Therefore, we partially specialize the whole class and define the get_aggregate_function anew every time.
```

Comment:
We are not sure what you mean by this, could you explain further?

Reply:
Sorry for my brevity.
I do not get what you mean with 'partially specialized'. Do you mean templated? Otherwise (or maybe both), you might be referring to the speciality of lambda classes to have a unique, unnamed type that makes it impossible to overload methods returning a lambda function as well as even just storing them into a common variable without wrapping them into a heavy-weight std::function? (Sorry for this sentence)

As you see, it just wasn't clear to me what exactly you mean, or did I miss something?

Reply:
> I do not get what you mean with 'partially specialized'.

My guess would be this: https://en.cppreference.com/w/cpp/language/partial_specialization